### PR TITLE
Update how we do tracing

### PR DESCRIPTION
**Pull request by svrnm**
Addresses problem (2) in #4107 by always including tracing.js in the build. This PR includes an additional features:

Users can run `npm run serve` with OTEL_EXPORTER_OTLP_ENDPOINT or OTEL_TRACES_EXPORTER set to either define their own OTLP endpoint or to write spans to the console, helps debugging that code

### CODE REVIEW

**Review by cartermp — APPROVED** [review]

**Review by theletterf — APPROVED** [review]

File: layouts/partials/hooks/body-end.html
Comment (chalin) [inline layouts/partials/hooks/body-end.html:23]: How about assuming production, and making it a little more dry? Something like this could replace lines 1 to 23, inclusive:

```html
<script>
  var tracingEnabled = true;
  var collectorType = 'otlp';
  var otlpTracesEndpointUrl = 'https://otelwebtelemetry.com/v1/traces';
{{ if eq (getenv "OTEL_TRACES_EXPORTER") "console" -}}
  collectorType = 'console';
{{ else if getenv "OTEL_EXPORTER_OTLP_ENDPOINT" -}}
  otlpTracesEndpointUrl = '{{ getenv "OTEL_EXPORTER_OTLP_ENDPOINT" }}';
{{ else if not hugo.IsProduction -}}
  tracingEnabled = false;
{{ end -}}
</script>
```

**Review by chalin — APPROVED** [review]
Nice fix. See inline for a suggested code refactoring. Otherwise, LGTM if y'all are happy!

File: layouts/partials/hooks/body-end.html
Comment (svrnm) [reply layouts/partials/hooks/body-end.html:23]: yeah, that's a good idea, let me do that.

Comment (svrnm) [reply layouts/partials/hooks/body-end.html]: applied

File: .github/workflows/check-links.yml
Comment (svrnm) [inline .github/workflows/check-links.yml:23]: This breaks if tracing.js is always included and by that always build, so we need to decide what is more important for us (not sure yet exactly on the overall performance impact)

Comment (svrnm) [reply .github/workflows/check-links.yml:23]: This is  blocking the PR to be merged, @open-telemetry/docs-approvers please take a look

**Pull request closed without merging**
